instance/tests/models/test_openedx_instance.py

```diff
@@ -733,9 +735,10 @@ def test_archive_no_active_appserver(self, mock_reconfigure, mock_disable_monito
         ])
         self.assertTrue(instance.ref.is_archived)
 
+    @patch('instance.models.server.OpenStackServer.public_ip')
```

Comment:
I don't understand immediately -- why is this needed now?

Reply:
the call to public_ip is an external call. On line 772, there's a call to
        # Set single app server active - this will be the reference date
        with freeze_time(reference_date):
            active_appserver.make_active()

so that will check the public_ip. Need to patch the public_ip call to avoid failures due to the external API call. Basically whereever the make_active() funciton is called, we need to patch the public_ip because we have added it as a precondition before making the appserver active.

Reply:
Well, I removed the mock and it passed anyway, so removing it now.

Reply:
:+1:

Reply:
actually it didn't pass. had to add it back. Sorry about the confusion.